skopt/tests/test_space.py

```diff
@@ -65,22 +65,40 @@ def test_integer():
     assert_array_equal(a.inverse_transform(random_values), random_values)
 
 
+def test_categorical_transform_binary():
+    categories = ["apple", "orange"]
+    cat = Categorical(categories)
+
+    apple = [0.]
+    orange = [1.]
+
+    assert_equal(cat.transform(["apple"]).size, cat.transformed_size)
+    assert_array_equal(cat.transform(categories), [apple, orange])
```

Comment:
We should change `Categorical` to handle this special case, such that this outputs `[[1.0, 0.0], [0.0, 1.0]]`. The reason `LabelBinarizer` handles it this way is because you can infer what the other column is given the output of `transform`.